api/src/routes/user.ts

```diff
@@ -301,5 +303,59 @@ export const userRoutes: FastifyPluginCallbackTypebox = (
     }
   );
 
+  fastify.post(
+    '/user/report-user',
+    {
+      schema: schemas.reportUser,
+      preValidation: (req, reply, done) => {
+        // NOTE: Is this the right hook?
```

Comment:
```suggestion
      preHandler: (req, _reply, done) => {
```
`preHandler` makes more sense to me, since we end up with cleaner code overall.